feature - add new day/night cycle example, add preview, update README.md

micarner opened the pull request:
Hey there, hopefully this is up to standard. I added a little slider so they can adjust day/night cycle.

vasturiano commented:
Much appreciated you putting together this PR.

So, the shader is phenomenal! 

However, I think the example would be even better it it uses the present time for plotting the position of the sun. Including the correct representation of the sun declination. Most of the time it's not directly above the equator but above or below (except at equinox). 

Much like in this example, which already includes all the calcs for the sun position:
https://globe.gl/example/solar-terminator/

Doable?

micarner commented:
That should be doable. Let me take a stab at it!

vasturiano commented:
@micarner I've refactored the example a bit and made it represent the solar declination according to the present time.
https://vasturiano.github.io/react-globe.gl/example/day-night-cycle/

I think it looks fantastic thanks to your shader, so thank you! 👍

micarner commented:
Np sir thank you for wrapping that up.  I'm sorry I meant to finish this
but an important job popped up and I had to put a lot of things on hold.

On Sun, Mar 2, 2025, 5:22 PM Vasco Asturiano ***@***.***>
wrote:

> @micarner <https://github.com/micarner> I've refactored the example a bit
> and made it represent the solar declination according to the present time.
> https://vasturiano.github.io/react-globe.gl/example/day-night-cycle/
>
> I think it looks fantastic thanks to your shader, so thank you! 👍
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vasturiano/react-globe.gl/pull/190#issuecomment-2692962545>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE32HKCB5CUWRI7ZRFAZZ2D2SOHCFAVCNFSM6AAAAABWWYNDUWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJSHE3DENJUGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: vasturiano]*vasturiano* left a comment
> (vasturiano/react-globe.gl#190)
> <https://github.com/vasturiano/react-globe.gl/pull/190#issuecomment-2692962545>
>
> @micarner <https://github.com/micarner> I've refactored the example a bit
> and made it represent the solar declination according to the present time.
> https://vasturiano.github.io/react-globe.gl/example/day-night-cycle/
>
> I think it looks fantastic thanks to your shader, so thank you! 👍
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vasturiano/react-globe.gl/pull/190#issuecomment-2692962545>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE32HKCB5CUWRI7ZRFAZZ2D2SOHCFAVCNFSM6AAAAABWWYNDUWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJSHE3DENJUGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
